### PR TITLE
Adds support of buildpackages in buildpack store and updates freezer

### PR DESCRIPTION
**Pull request by ForestEckhardt**
## Checklist

* [x] I have viewed, signed, and submitted the Contributor License Agreement.
* [ ] I have linked issue(s) that this PR should close using keywords or the Github UI (See [docs](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue))
* [ ] I have added an integration test, if necessary.
* [x] I have reviewed the [styleguide](https://github.com/paketo-buildpacks/community/blob/main/STYLEGUIDE.md) for guidance on my code quality.
* [x] I'm happy with the commit history on this PR (I have rebased/squashed as needed).

### CODE REVIEW

File: packagers/jam.go [hunk 1]
```diff
@@ -47,15 +68,32 @@ func (j Jam) Execute(buildpackDir, output, version string, offline bool) error {
 	args := []string{
 		"pack",
 		command, filepath.Join(buildpackDir, buildpackOrExtensionToml),
-		"--output", output,
+		"--output", filepath.Join(jamOutput, fmt.Sprintf("%s.tgz", version)),
```
Comment (robdimsdale) [inline packagers/jam.go:71]: can we encapsulate `fmt.Sprintf("%s.tgz", version)` into a variable - both for re-use but also to give it a useful name

File: packagers/jam_test.go [hunk 2]
```diff
@@ -16,12 +19,22 @@ func testJam(t *testing.T, context spec.G, it spec.S) {
...
+		packager packagers.Jam
 	)
 
 	it.Before(func() {
 		executable = &fakes.Executable{}
-		packager = packagers.NewJam().WithExecutable(executable)
+		pack = &fakes.Executable{}
+
+		tempOutput = func(string, string) (string, error) {
+			return "some-jam-output", nil
```
Comment (robdimsdale) [inline packagers/jam_test.go:34]: Does this work if the tests are run in parallel?

I was expecting to see `t.TempDir` as a testing substitute for the default `os.MkdirTemp`

**Review by robdimsdale — APPROVED** [review]
Generally looks good. A couple of non-blocking questions/comments.

File: packagers/jam_test.go (hunk 2 above)
Comment (ForestEckhardt) [reply packagers/jam_test.go:34]: I don't think I have ever seen issues with `os.MkdirTemp` and thread safety. It was just the function that I am most familiar with.

File: packagers/jam.go (hunk 1 above)
Comment (ForestEckhardt) [reply packagers/jam.go:71]: Sure I have no objections. Push something up soon.

**Review by robdimsdale — APPROVED** [review]

**Pull request merged**